packages/gatsby-remark-embed-snippet/src/__tests__/index.js

```diff
@@ -220,6 +247,40 @@ describe(`gatsby-remark-embed-snippet`, () => {
 
       expect(transformed).toMatchSnapshot()
     })
+
+    it(`should support hideline-range and hideline-next-line markers`, () => {
+      fs.readFileSync.mockReturnValue(
+        `
+        <!-- hideline-range{4,9-10} -->
```

Comment:
this is a little confusing - lines 9 and 10 are:
```
            <ul>
              <li>hidden</li>
```
and that's not exactly what is being hidden - both `<li>hidden</li>` are being hidden which are lines 10 and 11 (unless my math skills are worse that I thought :D)

Reply:
Do `<!-- hideline-range{4,9-10} -->` lines are being ommitted from line number calcs?

Reply:
@pieh I agree, this test case got a bit confusing. I'll improve this.

Reply:
@pieh I've improved some test cases, and I think it has now become more consistent.  Just as @bvaughn suggested:

```html
<html>
    <body>
    <p>
        <!-- hideline-range{1-2} -->
        <span>hidden</span>
        <span>hidden</span>
    </p>
    <ul>
        <!-- hideline-range{1-2} -->
        <li>hidden</li>
        <li>hidden</li>
        <li>not hidden</li>
    </ul>
    <!-- hideline-next-line -->
    <span>hidden</span>
    </body>
</html>
```

:smiley:

Reply:
Right, but that just hides (potential - not sure on that) issue - how it should behave when you have:
```html
<ul>
        <!-- hideline-range{1-2,6} -->
        <li>hidden</li>
        <li>hidden</li>
        <li>not hidden</li>
        <!-- hideline-next-line -->
        <li>hidden</li>
        <li>A: is this hidden?</li>
        <li>B: or this is this hidden?</li>
    </ul>
```
which line should be hidden with that - A or B? I think currently B will be hidden (that was the case in previous test), but intuitively I would expect A to be hidden I think

Reply:
@pieh I agree, but in this case this will be the output:

```html
<ul>
    <li>not hidden</li>
    <li>A: is this hidden?</li>
</ul>
```
This is because ```<!-- hideline-next-line -->``` directive is omitted from the line number calcs.

Although it seems a bit confusing, I think there are other ways to write the code to be consistent.

What do you think?

Reply:
@pieh It might be interesting to show in READ.ME more usage examples to avoid using directives inconsistently.

Reply:
Sorry for late response - I think if this behaviour is consistent with how highlighting calcs are done then that's ok and we shouldn't make changes to it. So README example explaining how it works would be enough. 

We can revisit it and potentially change that for next major release, but I'm not sure it's worth it.